src/analyzer.ts

```diff
@@ -203,16 +201,47 @@ export class Analyzer {
               col: scannedDependency.locationOffset.col,
               filename: document.url
             };
-            const scannedDocument = await this._analyzeSource(
-                scannedDependency.type, scannedDependency.contents,
-                document.url, locationOffset,
-                scannedDependency.attachedComment);
-            scannedDependency.scannedDocument = scannedDocument;
-            scannedDependency.scannedDocument.isInline = true;
-            return scannedDocument;
+            try {
+              const scannedDocument = await this._analyzeSource(
+                  scannedDependency.type, scannedDependency.contents,
+                  document.url, locationOffset,
+                  scannedDependency.attachedComment);
+              scannedDependency.scannedDocument = scannedDocument;
+              scannedDependency.scannedDocument.isInline = true;
+              return scannedDocument;
+            } catch (err) {
+              if (err instanceof WarningCarryingException) {
+                const e: WarningCarryingException = err;
+                e.warning.sourceRange =
+                    correctSourceRange(e.warning.sourceRange, locationOffset);
+                warnings.push(e.warning);
+                return null;
+              }
+              throw err;
+            }
           } else if (scannedDependency instanceof ScannedImport) {
-            const scannedDocument =
-                await this._analyzeResolved(scannedDependency.url);
+            let scannedDocument: ScannedDocument;
+            try {
+              // HACK(rictic): this isn't quite right either, we need to get
+              //     the scanned dependency's url relative to the basedir don't
+              //     we?
+              scannedDocument = await this._analyzeResolved(
+                  this._resolveUrl(scannedDependency.url));
+            } catch (error) {
+              if (error instanceof NoKnownParserError) {
+                // We probably don't want to fail when importing something
+                // that we don't know about here.
+                return null;
+              }
+              error = error || '';
+              warnings.push({
+                code: 'could-not-load',
+                message: `Unable to load import: ${error.message || error}`,
+                sourceRange: scannedDependency.sourceRange,
+                severity: Severity.ERROR
+              });
+              return null;
+            }
             scannedDependency.scannedDocument = scannedDocument;
             return scannedDocument;
           } else {
```

Comment:
This is dead code. Maybe rewrite the filter/map so you don't feel the need for it?